Matrix Zone diagram (multiple substations) : documentation update (#582)
powsybl/powsybl-diagram: single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/layout/zonebygrid/README.md

```java
    // Display substations on not empty Matrix cell
    matrix.stream().filter(c -> !c.isEmpty()).map(MatrixCell::graph).forEach(graph -> layoutBySubstation.get(graph).run(layoutParameters));
 ```
- each substation is moved into matrix position as specified

### Snakeline way computation between substation

The `Grid` class is a 2D array as a list of `Node` class representing each pixel of SLD output file.
Each `Node` store :
* a cartesian position (x and y)
* an availability (Node can be used are not)
* a walkthrough cost
* a parent node reference
* a distance to goal point

#### Exclusion area
An exclusion area is all `Node` with an availability equals to `false`
This area can be used to draw a `snakeline`.
This area allow snakeline to escape to:
- voltagelevels
- empty areas (missing substations)
- previous snakelines right angles

#### Shorter path computation
Dijkstra's computation steps:
* starting point cost is set to 0
* get nearest neighbors (left, right, up and down): no diagonal moves
  * these neighbors are used only if:
    * the neighbor is available
    * the neighbor was not already visited

>   * to avoid useless right angle the cost is increased when next point will create a right angle

Reviewer: ```suggestion
  * In order to avoid superfluous right angles, the cost is increased when the next point creates a right angle
```

> * In case of not route found, a strait line is drawn.

Reviewer: ```suggestion
* If no route can be computed by the algorithm, a straight line is drawn from the starting point to the end point of the snakeline (diagonal moves are allowed here)
```
